src/sentry/incidents/endpoints/organization_alert_rule_index.py

```diff
@@ -33,15 +45,72 @@ def post(self, request, organization):
+
+        alert_rule_serializer = AlertRuleSerializer(
+            context={"organization": organization, "access": request.access}, data=data
         )
+        if alert_rule_serializer.is_valid():
+            alert_rule = alert_rule_serializer.save()
+        else:
+            return Response(
+                {
+                    "AlertRuleError": alert_rule_serializer.errors,
+                    "message": "Could not create AlertRule",
+                },
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+        triggers = data.get("triggers", [])
+        for trigger in triggers:
+            # TODO: Implement trigger constraints for thresholds (warning triggers & resolves before critical)
+            trigger_serializer = AlertRuleTriggerSerializer(
+                context={
+                    "organization": organization,
+                    "alert_rule": alert_rule,
+                    "access": request.access,
+                },
+                data=trigger,
+            )
+            if trigger_serializer.is_valid():
+                trigger_obj = trigger_serializer.save()
+            else:
+                # Could not create alert rule trigger.
+                # TODO: ROLL BACK ALERT_RULE CREATION (and trigger if any yet). DELETE IT?
+                return Response(
+                    {
+                        "AlertRuleTriggerError": trigger_serializer.errors,
+                        "message": "Could not create AlertRuleTrigger",
+                    },
+                    status=status.HTTP_400_BAD_REQUEST,
+                )
+
+            actions = trigger.get("actions", [])
+            for action in actions:
+                action_serializer = AlertRuleTriggerActionSerializer(
+                    context={
+                        "organization": organization,
+                        "alert_rule": alert_rule,
+                        "trigger": trigger_obj,
+                        "access": request.access,
+                    },
+                    data=action,
+                )
 
-        if serializer.is_valid():
-            alert_rule = serializer.save()
-            return Response(serialize(alert_rule, request.user), status=status.HTTP_201_CREATED)
+                if action_serializer.is_valid():
+                    try:
+                        action = action_serializer.save()
+                    except InvalidTriggerActionError as e:
+                        # Could not create a trigger action
+                        # TODO: ROLL BACK ALL CREATIONS (Trigger(s), action(s) if any, and the alert rule)
+                        return Response(e.message, status=status.HTTP_400_BAD_REQUEST)
 
-        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+        # If we get here, the alert rule, triggers, and actions all created successfully. So return that.
+        return Response(serialize(alert_rule, request.user), status=status.HTTP_201_CREATED)
```

Comment:
Does this now need to return all the details we created? Probably worth checking with @billyvg

Reply:
What is our general API pattern here?

Reply:
We don't really have a pattern for this situation. Usually we create things individually and return those individual items. Just wasn't sure if you'd want to use the response to show the full rule once it's created?

Reply:
Unlikely, we generally redirect you back to the listing page after you create/edit something

Reply:
I think it's general web practice to return the object you created, right?

But we /could/ save some data and return just a 201 if this is the only use case there will ever be?

It does even seem to be our own standard to return objects after creation, and we have been returning all of this data + overhead with the multi-endpoint method anyways. I have no idea how much money/bandwidth we'd actually be saving if we made the 201, but if it's negligible, I say we return the object.

Reply:
Not really looking to save money/bandwidth, mostly just trying to reduce complexity, since if we want to serialize the whole tree then we need a detailed serializer to do that. I think it's probably fine to just serialize the `alert_rule` like we're doing here.